### DISCUSSION TITLE
Perimeter Overlap Expert Settings

### DISCUSSION THREAD

**Discussion by FozzyMaker** [supermerill/SuperSlicer#1800] [General]
Expert--Print Settings->Width & Flow->Overlap->Perimeter overlap

I didn't see a thread to report it's usefulness and so I wanted to comment on how I like these settings. 
I have only tried experimenting with the gap fill setting so far, but I do like it. I was able to reduce the amount of over extruded plastic for gap fill by reducing this parameter to 75%.
I'm going to work on some other settings before I play around with the default and external settings, but I wanted to make a comment about this setting so that it will stay in future versions of SuperSlicer.

**Comment by supermerill** [comment]
thank you, noted.

I think i can now remove the warning in the tooltip.
I'll adding the next version a new "global" overlap in the filament tab, as it seems that some filaments like abs need a lower global overlap (not only for perimeters but also for infill)

**Comment by DzzD** [comment]
I would just like to mention that those settings (if not set to zero) "Expert--Print Settings->Width & Flow->Overlap->***" are producing artefacts on the front of a simple Benchy print (at least with a cr10S Pro and defaults settings provided), I have tried to understand why and wich setting exacly but that's not that easy.

With all those settings off, the Benchy hull is perfecly smooth, but with default settings it show a lot of artifacts.

![image](https://user-images.githubusercontent.com/12198339/140604043-be28669a-e9e8-4686-a183-fa725b1c165f.png)

**Comment by adamrumjahn** [comment]
It seems like these overlap settings were optimized for PLA, however, I'm printing other materials like PEEK and the overlap settings are causing excess overextrusion and the final dimensions far out of tolerance. 

For example, a 2mm thick wall if sliced with 0% overlap or 100% overlap will have the same toolpath for the external perimeters, however, the spacing of the internal perimeters and infill will be such that there is so much over-extrusion that the wall becomes 2.5mm thick.  If I don't use this overlap setting, however, I do experience small gaps inside of the print and my goal is to achieve 100% density.  

In summary, the overlap feature is really great, but there needs to be a way to compensate for the overextrusion, perhaps by allowing a different flowrate setting for the external perimeter (that is less than the sum of all of the overlaps, or else adjustable by %).  Would this be possible?

**Reply by adamrumjahn** [reply]
Actually even better would be to change the tool path so that the "position" of the tool paths are moved closer together to compensate for the final thickness caused by the over-extrusion, while the "spacing" remains the same to still achieve the overlap (and fully dense internal perimeters/infill).

**Reply by supermerill** [reply]
> For example, a 2mm thick wall if sliced with 0% overlap or 100% overlap will have the same toolpath for the external perimeters, however, the spacing of the internal perimeters and infill will be such that there is so much over-extrusion that the wall becomes 2.5mm thick. 

a 100% overlap => no over or under-extrusion
a 0% overlap => under-extrusion

if you have over-extusion, it's not coming from the slicer but from your extrusion mult/filament/printer

you can set a max overlap In the filament setting

**Reply by jeedee23** [reply]
Use even-odd settings on the perimeter widths (new feature request), use extra perimeter on odd layers, problem totally gone. During a while now I always create 2 gcode files (I call them odd and even, even having a perimeter width of 20% less) with just the perimeter width set different  (and extra perimeter on odd layers), then I wrote a script to merge the two files into one, the result is astonishing, and the structural enhancement formidable when printing the merged file.
This way you can not have overextrusion and the 'overlap' option is obsolete.
Watch the below video how even and odd layers change their perimeter widths

https://user-images.githubusercontent.com/38980341/170856025-2cf2369d-c903-4153-8573-0913f843b554.mp4

**Reply by DzzD** [reply]
May you give a screenshot on which parameters you talk about odd/even ?

**Reply by jeedee23** [reply]
![image](https://user-images.githubusercontent.com/38980341/170858707-a6c3ec18-b230-4805-a50a-2fc73a6cf25c.png)

That one (and only that one of course) is 130 on my file called "even" and 110 on the file called "odd", the script then takes layer 0 from 'even', layer 1 from odd, layer 2 from 'even', etc .. resulting in exactly the same amount of layers, but the structural integrity is way better. I only do this with 2 perimeters now, for small objects, but the effect will be much bigger with bigger objects and 3 or 4 perimeters, since also the infill will shift.

the setting 'extra perimeter on odd' is already present:
![image](https://user-images.githubusercontent.com/38980341/170858843-765929dd-b511-41a3-b88f-792e62f434ac.png)

**Reply by jeedee23** [reply]
a better video screenshot, in the previous one the infill did not move:

https://user-images.githubusercontent.com/38980341/170859460-d9ec8938-2187-4d5b-813f-dd75cfc1ef73.mp4

**Reply by DzzD** [reply]
Ha ok, I like this technic, did not know about the extra-perimeter on odd, sur it will enforce the structural of the object, but still dont understand why you increase flow on even perimeter ? does odd-extra setting is not enought ?

**Comment by jeedee23** [comment]
Well, it's like building a (European) house, the bricks are shifted by each layer... the 'odd extra perimeter' does already a good job (for the shell), but if there was an even/odd perimeter width setting, it becomes impossible to peel layers off, meaning adhesion is everywhere... sometimes, with abs I had a layer peeling off due to warping: that is completely gone. Did not apply this method on difficult ABS or nylon parts which I sometimes print on a cold hot-glue bed (so: cold bed with frozen hot-glue, and raft is embedded in the hotglue, which freezes again, no warping, 100% adhesion to the bed. Just warm up the bed afterwards, remove the part, let cool down when flat and continue.  That, and glass and of course, if needed a hot chamber.
